pkg/mover/chart.go

```diff
@@ -131,48 +128,56 @@ type ChartMoveRequest struct {
 	Target Target
 }
 
-// ChartMover represents a Helm Chart moving relocation. It's initialization must be done view NewChartMover
-type ChartMover struct {
-	chartDestination        string
-	imageChanges            []*internal.ImageChange
-	chartChanges            []*internal.RewriteAction
-	sourceContainerRegistry internal.ContainerRegistryInterface
-	targetContainerRegistry internal.ContainerRegistryInterface
-	targetOfflineTarPath    string
-	chart                   *chart.Chart
-	logger                  Logger
-	retries                 uint
+// ChartData is the part of the Chart mover data that should be saved for a move to complete
+// without any network access to the original chart resources
+type ChartData struct {
+	chart        *chart.Chart
+	imageChanges []*internal.ImageChange
 	// raw contents of the hints file. Sample:
 	// test/fixtures/testchart.images.yaml
 	rawHints []byte
 }
 
+// ChartMover represents a Helm Chart moving relocation. It's initialization must be done view NewChartMover
+type ChartMover struct {
+	ChartData
+	chartDestination          string
+	chartChanges              []*internal.RewriteAction
+	sourceContainerRegistry   internal.ContainerRegistryInterface
+	targetContainerRegistry   internal.ContainerRegistryInterface
+	targetIntermediateTarPath string
+	logger                    Logger
+	retries                   uint
+}
+
 // NewChartMover creates a ChartMover to relocate a chart following the given
 // imagePatters and rules.
 func NewChartMover(req *ChartMoveRequest, opts ...Option) (*ChartMover, error) {
-	chart, err := loader.Load(req.Source.Chart.Local.Path)
-	if err != nil {
-		return nil, &ChartLoadingError{Path: req.Source.Chart.Local.Path, Inner: err}
-	}
-
-	if err := validateTarget(&req.Target); err != nil {
-		return nil, err
-	}
-
 	sourceAuth := req.Source.Containers.ContainerRepository
 	targetAuth := req.Target.Containers.ContainerRepository
 	cm := &ChartMover{
-		chart:                   chart,
 		logger:                  defaultLogger{},
 		retries:                 DefaultRetries,
 		sourceContainerRegistry: internal.NewContainerRegistryClient(sourceAuth),
 		targetContainerRegistry: internal.NewContainerRegistryClient(targetAuth),
 	}
-	if req.Target.Chart.Archive != nil {
-		cm.targetOfflineTarPath = req.Target.Chart.Archive.Path
+	if err := validateTarget(&req.Target); err != nil {
+		return nil, err
+	}
+	if req.Source.Chart.Local != nil {
+		err := cm.loadChart(req.Source.Chart.Local.Path, req.Source.ImageHintsFile)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, fmt.Errorf("must provide a local chart as input")
```

Comment:
why has this changed before we loaded and relied on the returned error. Is this to prepare to load an intermediate bundle?

Reply:
Yes